pennylane/_qubit_device.py

```diff
@@ -0,0 +1,282 @@
+            prob = np.abs(self._state ** 2)
+            prob = self.marginal_prob(prob, wires=wires)
+            return (eigvals @ prob).real
+
+        # estimate the ev
+        return np.mean(self.sample(observable))
+
+    def var(self, observable):
+        wires = observable.wires
+        self.rotate_basis(observable)
+
+        sampled_basis_states = self.sample_basis_states()
+        self._samples = self.convert_basis_states_to_binary(sampled_basis_states, self.num_wires)
+
+        if self.analytic:
+            # exact variance value
+            eigvals = observable.eigvals
+            prob = np.abs(self._state ** 2)
+            prob = self.marginal_prob(prob, wires=wires)
+            return (eigvals ** 2) @ prob - (eigvals @ prob).real ** 2
+
+        return np.var(self.sample(observable))
+
+    @staticmethod
+    def convert_basis_states_to_binary(basis_states, num_wires):
+        """Convert the basis states from base 10 to binary representation.
+
+        Args:
+            basis_states (list[int]): the list containing the basis states in base 10
+            num_wires (int): the number of wires for which we have the basis states
+
+        Returns:
+            array[float]: samples in an array of dimension ``(n, num_wires)``
+        """
+        return (((basis_states[:, None] & (1 << np.arange(2**num_wires)))) > 0).astype(int)
+
+#    def generate_samples(self):
+        # generate computational basis samples
+        # sample from the computational basis states based on the state probability
+
+
+    def sample_basis_states(self):
+        """Sample from the computational basis states based on the current state probability.
+
+        """
+        return np.random.choice(self._basis_states, self.shots, p=self._probability)
+
+    def sample(self, observable):
+        """Return a sample of an observable for software simulators.
+
+        The number of samples is determined by the value of ``Device.shots``,
+        which can be directly modified.
+
+        Note: all arguments support _lists_, which indicate a tensor
+        product of observables.
+
+        Args:
+            observable (str or list[str]): name of the observable(s)
+            wires (List[int] or List[List[int]]): subsystems the observable(s) is to be measured on
+            par (tuple or list[tuple]]): parameters for the observable(s)
+
+        Raises:
+            NotImplementedError: if the device does not support sampling
+
+        Returns:
+            array[float]: samples in an array of dimension ``(n, num_wires)``
+        """
+        if isinstance(observable, qml.Identity):
+            return np.ones([self.shots])
+
+        self.rotate_basis(observable)
+
+        sampled_basis_states = self.sample_basis_states()
+        self._samples = self.convert_basis_states_to_binary(sampled_basis_states, self.num_wires)
+
+        if isinstance(observable.name, str) and observable.name in {"PauliX", "PauliY", "PauliZ", "Hadamard"}:
```

Comment:
Now we could also replace this with the more succinct `not isinstance(observable, Tensor)`

Reply:
This would, however, leave out the ``Hermitian`` case, correct?